tools/autograd/gen_inplace_or_view_type.py

```diff
@@ -247,22 +285,139 @@ def emit_view_body(fn: NativeFunctionWithDifferentiabilityInfo, var: str) -> Tup
+
+    ret_and_arg_types = ', '.join([dispatcher_sig.returns_type()] + [a.type.cpp_type() for a in dispatcher_exprs])
+    # code-generated tracing kernels plumb and recompute dispatch keys directly through the kernel for performance.
+    # See Note [Plumbing Keys Through The Dispatcher] for details.
+    dispatch_key_set = 'ks & c10::DispatchKeySet(c10::DispatchKeySet::FULL_AFTER, c10::DispatchKey::InplaceOrView)'
+    redispatch_args = ', '.join([dispatch_key_set] + [a.expr for a in dispatcher_exprs])
+
+    # Note that this calls the slow, dispatching variants of manual_cpp_binding ops.
+    # We could probably work harder to ensure that the fast variants are called instead, but the perf benefit would be minimal.
+    sig_group = CppSignatureGroup.from_native_function(f, method=False, fallback_binding=f.manual_cpp_binding)
+    if sig_group.faithful_signature is not None:
+        api_name = sig_group.faithful_signature.name()
+    else:
+        api_name = sig_group.signature.name()
+    # inplace_view_body.append(THROW_IF_VARIABLETYPE_ON)
+    if modifies_arguments(f):  # inplace op
+        inplace_view_body.append(INPLACE_REDISPATCH.substitute(
+            api_name=api_name,
+            unpacked_args=redispatch_args,
+        ))
+        for r in cpp.return_names(f):
+            inplace_view_body.append(f'torch::autograd::increment_version({r});')
+    else:  # view op
+        inplace_view_body.append(VIEW_REDISPATCH.substitute(
+            assign_return_values='auto ' + TMP_VAR + ' = ',
+            api_name=api_name,
+            unpacked_args=redispatch_args,
+        ))
+        call, rhs_value = emit_view_body(fn, TMP_VAR)
+        inplace_view_body.append(call)
+        assert rhs_value is not None
+        inplace_view_body.append(
+            ASSIGN_RETURN_VALUE.substitute(return_values=tie_return_values(f), rhs_value=rhs_value))
+    if f.func.returns:
+        inplace_view_body.append(f'return {get_return_value(f)};')
+    return inplace_view_body
+
+@with_native_function
+def gen_formals(f: NativeFunction) -> str:
+    return ', '.join(
+        # code-generated autograd kernels plumb and recompute dispatch keys directly through the kernel for performance.
+        # See Note [Plumbing Keys Through The Dispatcher] for details.
+        ['c10::DispatchKeySet ks'] +
+        [f'{cpp.argument_type(a, binds="__placeholder__").cpp_type()} {a.name}'
+         for a in f.func.schema_order_arguments()]
+    )
+
+def inplace_or_view_method_definition(fn: NativeFunctionWithDifferentiabilityInfo) -> Optional[str]:
+    f = fn.func
+    if get_view_info(fn) is None and not modifies_arguments(f):
+        return None
+    return METHOD_DEFINITION.substitute(
+        return_type=cpp.returns_type(f.func.returns),
+        type_wrapper_name=type_wrapper_name(f),
+        formals=gen_formals(f),
+        type_definition_body=emit_inplace_or_view_body(fn),
+    )
+
+def modifies_arguments(f: NativeFunction) -> bool:
```

Comment:
nit: function ordering. This should be before the functions that are above and below I think to be consistent?